### ISSUE TITLE
Icons break, when the token is refreshed in a hidden iframe

### ISSUE THREAD

**Issue by pgrm** [angular/material2#11680]
#### Bug, ~~feature request, or proposal~~:

Sometimes we see for the icon text (for instance `add` or `file_download`) instead of the actual icon. This seems to happen after the `oidc-client` library triggers an invisible iframe refresh with the identity server loaded, to reload a new JWT.

Icons looked fine before, suddenly some of them are replaced with text + elements get duplicated. After resizing the page slightly, everything looks normal again.

#### What is the expected behavior?

For it to work normally all the time. Example how a page looks correctly:

![image](https://user-images.githubusercontent.com/713456/40986386-590b5f58-68e6-11e8-8fc9-6e1c513b647b.png)

#### What is the current behavior?

Sometimes (very hard to reproduce) it suddenly looks like this:

![image](https://user-images.githubusercontent.com/713456/40986432-7a9b0d58-68e6-11e8-9cb6-8e22420286c5.png)

after slight resizing of the window, it looks correctly again.

#### What are the steps to reproduce?

We had some ideas about library dependencies and not cleaned up `node_modules` folders, based on which we thought we could reproduce it, and thought we fixed it, but it keeps coming back, and now we don't have any idea how to reproduce it anymore.

If it happens, it seems to be only happening after the iframe for JWT token refresh is reloaded (we are using the library `oidc-client`) but just because the iframe was reloaded (happens every 5 minutes) doesn't mean, it will break. This is what such a reload looks like roughly:

![image](https://user-images.githubusercontent.com/713456/40987105-14f934c8-68e8-11e8-990c-6955bec2f4af.png)

This one broke it icons, but it looks the same if it doesn't break them.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?

- angular version 5.2.11
- material version 5.2.5
- OS: observed it on mac and windows
- TypeScript version 2.6.2
- browsers: observed it on latest firefox and latest chrome

#### Is there anything else we should know?

Maybe this is a known issue with icon fonts in general, which has a very easy fix, we just don't even know what to look for online / how to describe it.

**Comment by jelbourn** [comment]
If you're seeing the text instead of the icons, it means either:
* The icon font is not loaded, or
* The icon font css class isn't being applied to the `mat-icon` element

Are you able to determine which of those is occurring?

**Comment by pgrm** [comment]
@jelbourn as you can see, when the screen breaks, the elements with the broken icons get duplicated and the last element has the correct icon so ...

- the icon font is not loaded - I don't think this is possible, just clicking on the screen fixes it immediately (seemingly nothing happens according to the developer tools in chrome or firefox)
- the icon font css class isn't being applied - this might be some hint. Maybe the issue isn't the icon isn't displayed, but the actual issue is that those elemets get duplicated.

It can happen that elements get duplicated multiple times (while this bug is getting worse and worse) until the user clicks on the screen. At this time only the last element is rendered correctly. All other elements are rendered without the specific styles being applied. I also just noticed on the screenshot I provided, that the broken `Export` button is in black, while the correct one is orange. So it doesn't seem to be just the icons, but the whole CSS breaking. - Thx for the hint, we never noticed that before, because we were so fixated on the icons.

So probably the questions should change more to:

> Would you have any idea what could cause some elements to be duplicated?

**Comment by jelbourn** [comment]
If you're getting just regular elements duplicated, it doesn't seem like anything related to Angular Material. I'd suggest looking at whether some async operation in your app is escaping the Angular zone.

**Comment by pgrm** [comment]
@jelbourn yeah sry, we discussed it internally, and we do have a problem with the oidc library and e2e tests as well, because it seems to be constantly running an async loop in the background which means when our protractor tests are waiting for angular to complete they never execute.

Therefore your hint with the async operations and Angular zone seems a reasonable explanation (although really hard to verify - at least for me).

Thank you
